### PR TITLE
Rust Simple implementaition

### PR DESCRIPTION
**Pull request by Blui42**
## Description

This simple rust implementaion is very strongly oriented towards the original code.

Expected output is:
- std-err: Author; Passes; Time; Threads
- std-out: Blui42; (passes); (time); 1
- std-err: (passes/time) Passes per second
- std-err: For Validation: (count) Primes were found
If further validation is needed, line 61 can be uncommented to print all the primes to std-err.

## Contributing requirements

* [x] I read the contribution guidelines in CONTRIBUTING.md.
* [x] I placed my solution in the correct category folder.
* [x] I added a README.md or Dockerfile.
* [x] I selected `drag-race` as the target branch.
* [x] All code herein is licensed compatible with BSD-3.

### CODE REVIEW

**Comment by Azgrom** [comment]
You see, there was another solution_2 going on at #150...

Even so, your code is quite straightforward. I would only make the printed results more clear, and separate declaration and definition concerns in a different file.

```rust
$ cargo run --release
   Compiling primes v0.1.0 (/home/rafael/Documents/Repos/Blui/PrimeRust/solution_2)
    Finished release [optimized] target(s) in 1.08s
     Running `target/release/primes`
Blui42; 2532; 5.000130272; 1
78498
```
For example. I am assuming `2532` is the number of passes. But I also coded this exercise. Other person might not be able to infer it.

**Comment by marghidanu** [comment]
I think just renaming this to solution_3 should be enough to resolve this conflict. The output format is documented in CONTRIBUTING.md, this is to facilitate the automated reporting script to parse the output.

**Comment by Blui42** [comment]
- Changed name from solution_2 to solution_3
- Clarified output in std-err

New output:
```
Author; Passes; Time; Threads
Blui42; 4032; 5.000549; 1
806.31 Passes per second
For Validation: 78498 Primes were found
```

I cannot change `Blui42; 4032; 5.000549; 1` as it is required by this rule:

> The test code outputs the following text to standard output:
> 
> `<name>;<iterations>;<total_time>;<num_threads>`

File: PrimeRust/solution_3/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            }
+            for num in (factor*factor..=self.size).step_by(factor*2){
+                self.bits[num] = false;  // mark all multiples of this number as not prime
+            }
+            factor += 2;
+        }
+    }
+    pub(crate) fn print_results(&self, duration: f64, passes: u32){
+        eprintln!("Author; Passes; Time; Threads");  
+        println!("Blui42; {}; {}; 1", passes, duration);  // print required output to standard out
```
Comment (rhbvkleef) [inline PrimeRust/solution_3/src/main.rs]: Ideally, it should be this:

```suggestion
        println!("Blui42;{};{};1", passes, duration);  // print required output to standard out
```

**Comment by rhbvkleef** [comment]
I want to either see that a Dockerfile is added here, or wait for #168 and add a Dockerfile from it.

**Comment by rbergen** [comment]
I think either Dockerfile in the existing Rust solutions are a good starting point for this solution; it should not be difficult to create a working Dockerfile for this solution starting with either of those. 

@Blui42: will you add one?

**Comment by Blui42** [comment]
After getting frustrated trying to read the docs of docker for way longer than I should, I realized I only need to add a dockerfile, not a docker build.

Oh well...

**Comment by rbergen** [comment]
It seems like the Docker run image can't find/copy the executable it's looking for. Could it be that it's called "primes" instead of "prime_rust", in line with the package name you specified in Cargo.toml?

**Comment by Blui42** [comment]
Oh, right. Docker for some reason is not recognized as installed even though it is, so I had no way of testing. Will be fixed in less than five minutes.

File: PrimeRust/solution_3/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Rust solution by Blui42
+
+Faithful
```
Comment (rbergen) [inline PrimeRust/solution_3/README.md]: Please replace "Faithful" with:
```
![Category](https://img.shields.io/badge/Category-faithful-green) 
```

**Comment by rbergen** [comment]
I can confirm that the Dockerfile builds and runs. I found one last tiny thing in the README.md that I just commented on. Once that's in place I think we can merge this.

**Review by rbergen — APPROVED** [review]
@rhbvkleef or @marghidanu To me this looks good to merge. Please merge if you agree.

**Pull request merged**